read_terraform_state_role.tf

```
@@ -0,0 +1,26 @@
+# ------------------------------------------------------------------------------
+# Create the IAM policy and role that allows read-only access to the Terraform
+# state for this project in the S3 bucket where Terraform remote state is
+# stored.
+# ------------------------------------------------------------------------------
+
+module "read_terraform_state" {
+  source = "github.com/cisagov/terraform-state-read-role-tf-module"
+
+  providers = {
+    aws       = aws.terraform
+    aws.users = aws.users
+  }
+
+  account_ids = [local.users_account_id]
+  role_name   = var.read_terraform_state_role_name
+  # It makes no sense to associate a "Workspace" tag with the
+  # Terraform read role, since it can read the state from any
+  # workspace.
+  #
+  # Such a tag will also flip flop as one switched from staging to
+  # production or vice versa, which is highly annoying.
+  role_tags                   = { for k, v in var.tags : k => v if k != "Workspace" }
+  terraform_state_bucket_name = "cisa-cool-terraform-state"
+  terraform_state_path        = "cool-images-assessment-images/*.tfstate"
```

Can we just make this `cool-images-assessment-images/terraform.tfstate`?  Or is there some benefit to using `*.tfstate` here that I'm not thinking of?
```suggestion
  terraform_state_path        = "cool-images-assessment-images/terraform.tfstate"
```

---

The other COOL repositories I looked at do the same thing so I was following suite. I can change it, but I'd like clarity on when it should be a wildcard and when it shouldn't.

---

This _probably_ started because the first repo from which we read remote state was [cisagov/cool-accounts](https://github.com/cisagov/cool-accounts), where this sort of voodoo would make sense.

Eventually even [cisagov/cool-accounts](https://github.com/cisagov/cool-accounts) will be split into separate repos, at which time the wildcard will no longer make sense even for it.

We definitely need to get some of this stuff written down in the style guide.

---

I think most of our use cases can avoid using `*.tfstate` since we very rarely have different TF state objects in the same subdirectory.  I think the `*.tfstate` should only be used when there is a chance that multiple TF states will be in use- but we tend to not do that anywhere.

---

After discussion this makes sense. I've limited the scope in a85157a.

---

Spiffy.